crates/plugins/src/version_check.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{
+    get_manifest_file_name, plugin_manifest::PluginManifest, PLUGIN_MANIFESTS_DIRECTORY_NAME,
+};
+use anyhow::{anyhow, Result};
+use semver::{Version, VersionReq};
+use std::{fs::File, path::Path};
+
+/// Checks whether the plugin supports the currently running version of Spin.
+pub fn assert_supported_version(spin_version: &str, supported: &str) -> Result<()> {
```

Comment:
"assert" is a very "should never happen" kind of name *grin*

Reply:
changed it to a nice kind `check_`